src/components/MenuBar/useToggleMenuList.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useLayoutEffect, useRef, useState } from 'react';
+
+const MENUBAR_PADDING_X = 8;
+
+const useToggleMenuList = () => {
+  const menuListRef = useRef<HTMLUListElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [menuListPosition, setMenuListPosition] = useState({ x: 0, y: 0 });
+
+  const toggleMenuList = (rect: DOMRect) => {
+    setIsOpen((prevOpenState) => !prevOpenState);
+
+    const x = rect.x;
+    const y = rect.y + rect.height;
+
+    setMenuListPosition({ x, y });
+  };
+
+  useEffect(() => {
+    const handleClickAway = (event: Event) => {
+      const menuListElement = menuListRef.current;
+      const isClickMenuBarItemAway =
+        event.target instanceof HTMLElement &&
+        menuListElement &&
+        menuListElement.previousSibling &&
+        !menuListElement.contains(event.target) &&
+        !menuListElement.previousSibling.isSameNode(event.target) &&
+        !menuListElement.previousSibling.contains(event.target);
+
+      if (isClickMenuBarItemAway) {
+        setIsOpen(false);
+      }
```

Comment:
menuList 컴포넌트 바깥을 클릭했는지 체크하는 조건문이 복잡해요. 조건문을 간소화할 수 있는 방법이 있을까요? 🤔

Reply:
아래 제안드린 것 처럼 수정하시면 복잡한 조건문을 완화할 수 있습니다!

```suggestion
const handleClickAway = (event: Event) => {
      const menuListElement = menuListRef.current;
      const isClickMenuBarItemAway = event.target instanceof HTMLElement && !menuListElement?.contains(event.target);

      if (isClickMenuBarItemAway) {
        setIsOpen(false);
      }
    };

    document.addEventListener('mousedown', handleClickAway);

    return () => {
      document.removeEventListener('mousedown', handleClickAway);
    };
```

다만 동일한 메뉴를 클릭했을 때 다시 팝업되는 이슈가 있으므로, 조건을 적절히 추가하여 해결하시면 될 것 같습니다. 😃

Reply:
@jaeung-E 처음에 23번 라인처럼 코드를 작성했는데 안 됐던 이유가 `click` 이벤트를 사용했기 때문이군요 ...! 감사합니다 😄